x/exchange/keeper/swap.go

```diff
@@ -68,18 +68,17 @@ func (k Keeper) SwapExactAmountIn(
 			return output, nil, err
 		}
 		output = res.Received
-		if !res.FullyExecuted {
-			if res.Paid.IsLT(currentIn) {
-				return output, nil, sdkerrors.Wrapf(
-					types.ErrSwapNotEnoughLiquidity, "in market %d; paid %s < input %s", marketId, res.Paid, currentIn)
-			}
+		if !full && res.Paid.IsLT(currentIn) {
+			return output, nil, sdkerrors.Wrapf(
+				types.ErrSwapNotEnoughLiquidity, "in market %d; paid %s < input %s", marketId, res.Paid, currentIn)
 		}
 		results = append(results, types.SwapRouteResult{
 			MarketId:         marketId,
 			ExecutedQuantity: res.ExecutedQuantity,
 			Input:            currentIn,
 			Output:           output,
-			Fee:              res.Fee,
+			// TODO: add Paid field?
+			Fee:              res.FeePaid,
 		})
 		currentIn = output
 	}
```

Comment:
In Line 118, what if the market has no order? Do we need to skip it if the market has no order?

Reply:
Markets with no limit orders will naturally be passed in the actual swap algorithm.